insights/combiners/os_release.py

```diff
@@ -0,0 +1,136 @@
+"""
+OSRelease
+=========
+The ``OSRelease`` combiner uses the following parsers to try to identify if the
+current host is installed with a "Red Hat Enterprise Linux" system.
+
+    - :py:class:`insights.parsers.os_release.OsRelease`
+    - :py:class:`insights.parsers.redhat_release.RedhatRelease`
+    - :py:class:`insights.parsers.installed_rpms.InstalledRpms`
+
+It provides an attribute `is_rhel` that indicates if the host is "RHEL" or not.
+It also provides an attribute `product` which returns the estimated OS name of
+the system, "Unknown" will be returned by default when cannot identify the OS.
+"""
+from insights.core.plugins import combiner
+from insights.parsers.installed_rpms import InstalledRpm, InstalledRpms
+from insights.parsers.os_release import OsRelease
+from insights.parsers.redhat_release import RedhatRelease
+from insights.parsers.uname import Uname
+
+RHEL_STR = "Red Hat Enterprise Linux"
+RHEL_KEYS = ['rhel', 'red hat enterprise linux']
+"""Keywords in 'os-release' or 'redhat-release' that indicates an RHEL system."""
+NON_RHEL_KEYS = ['oracle', 'suse', 'sles', 'fedora', 'centos']
+"""Keywords in 'os-release' or 'redhat-release' that indicates a NON-RHEL system."""
+MIN_RHEL_PKGS = [
+    'audit-libs',
+    'basesystem',
+    'bash',
+    'coreutils',
+    'dbus',
+    'dmidecode',
+    'dracut',
+    'filesystem',
+    'firewalld',
+    'glibc',
+    'gmp',
+    'krb5-libs',
+    'libacl',
+    'libgcc',
+    'libselinux',
+    'NetworkManager',
+    'openssl-libs',
+    'passwd',
+    'systemd',
+    'util-linux',
+    ('dnf', 'yum'),
```

Comment:
Do not need this tuple and just keep all as string.  The code of calculation is checking package existing or not.

Reply:
Nice catch, the tuple is for the former version which didn't calculate the "installed_packages". but with the introduction of the "installed_packages" the tuples are not necessary anymore. I'll expand them and simplify the code.

Reply:
updated